### ISSUE TITLE
Syntax highlighting gets confused if return type ends with a number

### ISSUE THREAD

**Issue by aeschli** [Microsoft/TypeScript-TmLanguage#207]
_From @kayahr on July 4, 2016 11:9_

- VSCode Version: 1.2.1
- OS Version: Debian GNU/Linux 8.5

Steps to Reproduce:

1. Open new TypeScript file.
2. Copy/Paste the following code:

```javascript
class Vector2 {
    public foobar(): Vector2 {
        return this;
    }
}

class Vector2D {
    public foobar(): Vector2D {
        return this;
    }
}
```

The method in the class `Vector2D` is properly colored but in the class `Vector2` the syntax highlighting gets confused. Looks like this has something to do with the class name ending with a number. 

Here is a screenshot showing a wrong color of the `return` and `this`keywords:

![highlighting](https://cloud.githubusercontent.com/assets/580493/16558618/28846992-41e7-11e6-90c5-9a74362ed30d.png)

In larger source code this gets quite confusing because subsequent methods in the same class are also affected.

_Copied from original issue: Microsoft/vscode#8720_

**Comment by kayahr** [comment]
By the way: The same problem can be reproduced right here on Github when formatting source code with language=typescript:

```typescript
class Vector2 {
    public foobar(): Vector2 {
        return this;
    }
}

class Vector2D {
    public foobar(): Vector2D {
        return this;
    }
}
```

**Comment by glen-84** [comment]
Duplicate of #188 and #153.

**Comment by sheetalkamat** [comment]
Fixed by #257
